starter-cli/src/main/java/io/micronaut/starter/cli/command/CreateChatBotBuilderCommand.java

```diff
@@ -114,21 +115,36 @@ enum ChatBotType {
 
     enum ChatBotDeployment {
 
-        LAMBDA("AWS Lambda", "Deploy to AWS Lambda", TelegramAwsChatBot.NAME, ApplicationType.FUNCTION),
-        AZURE("Azure Function", "Deploy to Azure as a Function", TelegramAzureChatBot.NAME, ApplicationType.FUNCTION),
-        GCP("Google Cloud Function", "Deploy to Google Cloud as a Function", TelegramGcpChatBot.NAME, ApplicationType.FUNCTION),
-        HTTP("HTTP", "Deploy as an HTTP Server", TelegramHttpChatBot.NAME, ApplicationType.DEFAULT);
+        LAMBDA("AWS Lambda", "Deploy to AWS Lambda", BasecampAwsChatBot.NAME, TelegramAwsChatBot.NAME, ApplicationType.FUNCTION),
+        AZURE("Azure Function", "Deploy to Azure as a Function", BasecampAzureChatBot.NAME, TelegramAzureChatBot.NAME, ApplicationType.FUNCTION),
+        GCP("Google Cloud Function", "Deploy to Google Cloud as a Function", BasecampGcpChatBot.NAME, TelegramGcpChatBot.NAME, ApplicationType.FUNCTION),
+        HTTP("HTTP", "Deploy as an HTTP Server", BasecampHttpChatBot.NAME, TelegramHttpChatBot.NAME, ApplicationType.DEFAULT);
 
         private final String title;
         private final String description;
-        private final String feature;
+        private final String basecampFeature;
+        private final String telegramFeature;
         private final ApplicationType applicationType;
 
-        ChatBotDeployment(String title, String description, String feature, ApplicationType applicationType) {
+        ChatBotDeployment(
+                String title,
+                String description,
+                String basecampFeature,
```

Comment:
can we use a list or a `Map<ChatBotType, String>` so that each time we add a new chatbot we don't have to modify the constructor?

Reply:
Switched to accept a function, this means we can use the new Java 17 switch construct and make sure we don't miss any

https://github.com/micronaut-projects/micronaut-starter/pull/2263/commits/a4a7d886109e6aa0bf1d2920cef70f153751c2f3

What do you think?